### PR TITLE
Fix broken syntax highlighting after the first submodule

### LINKED ISSUE #284: Broken syntax highlighting after the first submodule
```
---- MODULE foo ----
---- MODULE bar ----

Op1 == TRUE
====

Op2 == TRUE
====
```

Highlighting will not cover `Op2`.

### PR DESCRIPTION
**Pull request by FedericoPonzi**
Fixes #284 
image
image
image

I was also able to verify the update with a test.

### CODE REVIEW

File: languages/tlaplus-grammar.json [hunk 1]
```diff
@@ -170,6 +170,9 @@
                 }
             },
             "patterns": [
+                {
```
Comment (lemmy) [inline languages/tlaplus-grammar.json:173]: I know little about VSCode.  Is this what fixes the syntax highlighting?

Comment (FedericoPonzi) [reply languages/tlaplus-grammar.json:173]: Yes. As you mentioned in the issue, a regex is being used to identify a module but there are actually two regexes, one for the start token and one for the end (see line 151 and 166). This section define the patterns that can be found inside the body, by adding #module I'm allowing recursively searching for modules inside the module.

**Review by lemmy — APPROVED** [review]
LGTM

File: languages/tlaplus-grammar.json (hunk 1 above)
Comment (lemmy) [reply languages/tlaplus-grammar.json:173]: Could you add that explanation to the git commit message?

**Comment by FedericoPonzi** [comment]
I've reworded the commit message, and pushed an additional commit to fix the warning I've forgot to push in the other pr.

**Pull request merged**